src/javascripts/components/search.js

```diff
@@ -72,8 +72,28 @@ var AppSearch = {
     // add rest of the data here to build the item
     if (result) {
       var elem = document.createElement('span')
-      elem.textContent = result.title
-
+      var resultTitle = result.title
+      elem.textContent = resultTitle
+      if (result.aliases) {
+        var aliases = result.aliases.split(',')
+        var aliasesContainer = document.createElement('span')
+        var openBracket = document.createTextNode(' (')
+        var closeBracket = document.createTextNode(')')
+        // only show a matching alias if there are no matches in the title
+        if (resultTitle.toLowerCase().indexOf(searchQuery) === -1) {
+          var matchedAliases = aliases.filter(function (alias) {
+            return alias.indexOf(searchQuery) !== -1
+          }).map(function (item) {
+            return item.trim()
+          })
+          aliasesContainer.textContent = matchedAliases.join(', ')
+          if (matchedAliases.length > 0) {
+            aliasesContainer.insertBefore(openBracket, aliasesContainer.firstChild)
+            aliasesContainer.insertBefore(closeBracket, aliasesContainer.nextSibling)
+          }
+        }
+        elem.appendChild(aliasesContainer)
```

Comment:
This logic looks pretty complicated, should we have tests for the these different major forks in the logic?

Reply:
added a test for this